src/spikeinterface/exporters/to_ibl.py

```diff
@@ -0,0 +1,309 @@
+import numpy.typing as npt
+from tqdm.auto import tqdm
+
+from spikeinterface.core import ChannelSparsity, SortingAnalyzer
+from spikeinterface.core.template_tools import get_template_extremum_channel
+from spikeinterface.exporters import (
+    export_to_phy,
+)
+from spikeinterface.exporters.to_ibl_utils import (
+    WindowGenerator,
+    fscale,
+    hp,
+    rms,
+    save_object_npy,
+)
+
+
+def export_to_ibl(
+    analyzer: SortingAnalyzer,
+    output_folder: str | Path,
+    rms_win_length_sec=3,
+    welch_win_length_samples=1024,
+    total_secs=100,
+    only_ibl_specific_steps=False,
+    compute_pc_features: bool = False,  # shouldn't need these?
+    compute_amplitudes: bool = True,
+    sparsity: Optional[ChannelSparsity] = None,
+    copy_binary: bool = True,
+    remove_if_exists: bool = False,
+    template_mode: str = "median",
+    dtype: Optional[npt.DTypeLike] = None,
+    verbose: bool = True,
+    use_relative_path: bool = False,
+    **job_kwargs,
+):
+    """
+    Exports a sorting analyzer to the IBL gui format (similar to the Phy format with some extras).
+
+    Parameters
+    ----------
+    analyzer: SortingAnalyzer
+        The sorting analyzer object.
+    output_folder: str | Path
+        The output folder where the phy template-gui files are saved
+    rms_win_length_sec: float, default: 3
+        The window length in seconds for the RMS calculation.
+    welch_win_length_samples: int, default: 1024
+        The window length in samples for the Welch method.
+    total_secs: int, default: 100
+        The total number of seconds to use for the spectral density calculation.
+    only_ibl_specific_steps: bool, default: False
+        If True, only the IBL specific steps are run (i.e. skips calling `export_to_phy`)
+    compute_pc_features: bool, default: False
+        If True, pc features are computed
+    compute_amplitudes: bool, default: True
+        If True, waveforms amplitudes are computed
+    sparsity: ChannelSparsity or None, default: None
+        The sparsity object (currently only respected for phy part of the export)
+    copy_binary: bool, default: True
+        If True, the recording is copied and saved in the phy "output_folder"
+    remove_if_exists: bool, default: False
+        If True and "output_folder" exists, it is removed and overwritten
+    template_mode: str, default: "median"
+        Parameter "mode" to be given to WaveformExtractor.get_template()
+    dtype: dtype or None, default: None
+        Dtype to save binary data
+    verbose: bool, default: True
+        If True, output is verbose
+    use_relative_path : bool, default: False
+        If True and `copy_binary=True` saves the binary file `dat_path` in the `params.py` relative to `output_folder` (ie `dat_path=r"recording.dat"`). If `copy_binary=False`, then uses a path relative to the `output_folder`
+        If False, uses an absolute path in the `params.py` (ie `dat_path=r"path/to/the/recording.dat"`)
```

Comment:
```suggestion
    ----------
    analyzer : SortingAnalyzer
        The sorting analyzer object.
    output_folder : str | Path
        The output folder where the phy template-gui files are saved
    rms_win_length_sec : float, default: 3
        The window length in seconds for the RMS calculation.
    welch_win_length_samples : int, default: 1024
        The window length in samples for the Welch method.
    total_secs : int, default: 100
        The total number of seconds to use for the spectral density calculation.
    only_ibl_specific_steps : bool, default: False
        If True, only the IBL specific steps are run (i.e. skips calling `export_to_phy`)
    compute_pc_features : bool, default: False
        If True, pc features are computed
    compute_amplitudes : bool, default: True
        If True, waveforms amplitudes are computed
    sparsity : ChannelSparsity or None, default: None
        The sparsity object (currently only respected for phy part of the export)
    copy_binary : bool, default: True
        If True, the recording is copied and saved in the phy "output_folder"
    remove_if_exists : bool, default: False
        If True and "output_folder" exists, it is removed and overwritten
    template_mode : str, default: "median"
        Parameter "mode" to be given to WaveformExtractor.get_template()
    dtype : dtype or None, default: None
        Dtype to save binary data
    verbose : bool, default: True
        If True, output is verbose
    use_relative_path : bool, default: False
        If True and `copy_binary=True` saves the binary file `dat_path` in the `params.py` relative to `output_folder` (ie `dat_path=r"recording.dat"`). If `copy_binary=False`, then uses a path relative to the `output_folder`
        If False, uses an absolute path in the `params.py` (ie `dat_path=r"path/to/the/recording.dat"`)
```

We are working on implementing numpydoc linting in the future and it requires there to be a space after the parameter.